### PR TITLE
Show the proper auth identifier when listing mounts as json

### PR DESCRIPTION
**Pull request by icewind1991**
Also adjusted the import code to handle the change in output format

cc @PVince81

### CODE REVIEW

**Comment by icewind1991** [comment]
Fixes https://github.com/owncloud/core/issues/21771

**Comment by PVince81** [comment]
Code looks good :+1:

**Comment by DeepDiver1975** [comment]
no unit tets to cover this change?

**Comment by Xenopathic** [comment]
:+1:

**Comment by PVince81** [comment]
> no unit test to cover this change?

Good point. @icewind1991 please add a unit test.

**Comment by icewind1991** [comment]
I'm not sure how to make a meaningful unit tests for this change

**Comment by Xenopathic** [comment]
@icewind1991 Perhaps two auth mechanisms with the same scheme but different identifiers, make sure the export shows distinct values?

File: apps/files_external/command/listcommand.php [hunk 1]
```diff
@@ -151,7 +151,7 @@ public function listMounts($userId, array $mounts, InputInterface $input, Output
 					$config->getId(),
 					$config->getMountPoint(),
 					$config->getBackend()->getStorageClass(),
-					$config->getAuthMechanism()->getScheme(),
+					$config->getAuthMechanism()->getIdentifier(),
```
Comment (PVince81) [inline apps/files_external/command/listcommand.php]: maybe move this bigger block into a function and write a test for that function ?

**Comment by icewind1991** [comment]
Added test

**Comment by PVince81** [comment]
:+1:

**Comment by PVince81** [comment]
@DeepDiver1975 the unit test was added, merge ?

**Comment by icewind1991** [comment]
@DeepDiver1975 @LukasReschke @Xenopathic please review

**Comment by DeepDiver1975** [comment]
:+1:

**Pull request merged**